src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -665,84 +668,76 @@ async function fetchAndUseVersions() {
+
+  // These are the corresponding left/right <dialog> elements, which are empty
+  // until the user clicks the hamburger icon
+  const primaryDialog = document.getElementById("pst-primary-sidebar-modal");
+  const secondaryDialog = document.getElementById(
+    "pst-secondary-sidebar-modal",
   );
-  const primarySidebar = document.querySelector(".bd-sidebar-primary");
-  const secondarySidebar = document.querySelector(".bd-sidebar-secondary");
-
-  // Toggle buttons -
-  //
-  // These are the hamburger-style buttons in the header nav bar. When the user
-  // clicks, the button transmits the click to the hidden checkboxes used by the
-  // CSS to control whether the sidebar is open or closed.
-  const primaryClickTransmitter = document.querySelector(".primary-toggle");
-  const secondaryClickTransmitter = document.querySelector(".secondary-toggle");
+
+  // These are the hamburger-style buttons in the header nav bar. They only
+  // appear at narrow screen width.
+  const primaryToggle = document.querySelector(".primary-toggle");
+  const secondaryToggle = document.querySelector(".secondary-toggle");
+
+  // Cut nodes and classes from `from`, paste into/onto `to`
+  const cutAndPasteNodesAndClasses = (from, to) => {
+    Array.from(from.childNodes).forEach((node) => to.appendChild(node));
+    Array.from(from.classList).forEach((cls) => {
+      from.classList.remove(cls);
+      to.classList.add(cls);
+    });
+  };
+
+  // Hook up the ways to open and close the dialog
   [
-    [primaryClickTransmitter, primaryToggle, primarySidebar],
-    [secondaryClickTransmitter, secondaryToggle, secondarySidebar],
-  ].forEach(([clickTransmitter, toggle, sidebar]) => {
-    if (!clickTransmitter) {
+    [primaryToggle, primaryDialog, primarySidebar],
+    [secondaryToggle, secondaryDialog, secondarySidebar],
+  ].forEach(([toggleButton, dialog, sidebar]) => {
+    if (!toggleButton || !dialog || !sidebar) {
       return;
     }
-    clickTransmitter.addEventListener("click", (event) => {
+
+    // Clicking the button can only open the sidebar, not close it.
+    // Clicking the button is also the *only* way to open the sidebar.
+    toggleButton.addEventListener("click", (event) => {
       event.preventDefault();
       event.stopPropagation();
-      toggle.checked = !toggle.checked;
-
-      // If we are opening the sidebar, move focus to the first focusable item
-      // in the sidebar
-      if (toggle.checked) {
-        // Note: this selector is not exhaustive, and we may need to update it
-        // in the future
-        const tabStop = sidebar.querySelector("a, button");
-        // use setTimeout because you cannot move focus synchronously during a
-        // click in the handler for the click event
-        setTimeout(() => tabStop.focus(), 100);
-      }
+
+      // When we open the dialog, we cut and paste the nodes and classes from
+      // the widescreen sidebar into the dialog
+      cutAndPasteNodesAndClasses(sidebar, dialog);
```

Comment:
I realized that I should explain why I decided to take this approach. 

First, some background. When the website is wide, the sidebars are displayed directly on the page as sidebars. When the website is narrow, some CSS kicks in to make the sidebars into invisible drawers, or overlays, that become visible when the user clicks a button. A nice feature of the way this was implemented is that both the sidebars and the drawers share the same exact HTML. In other words, the HTML is not duplicated on the page so that you have one HTML tree for the sidebar and another HTML tree for the drawer.

For the mobile presentation of the sidebar content, I want to use the native HTML dialog element because it comes with all of the modal functionality that we want built in, which means less JavaScript code to write and maintain on our end. However, it would be incorrect to use the dialog element for the desktop presentation of the same content because it is not in a dialog, which is a descendant window of the primary window—it is in a sidebar.

That means that we have to have two different root nodes for the sidebar:

- Mobile: `<dialog>`
- Desktop: `<div>`

One possible solution would be to render the Jinja templates twice, once in the dialog element and once in the div. However, this increases the byte size of the HTML page and it creates potential HTML element id conflicts. 

So that's why I chose this implementation in which I move the sidebar content nodes from the div to the dialog, when the user clicks the button to open the mobile dialog, and then move them back when the user closes the mobile dialog.

Reply:
The only things that I'm worried about are: 
 - events, 
 - screen reader. 

I don't think that should have an impact on screen reader as they probably don't resize to mobile format. 

Events I don't know but I guess we'll see